core/common/src/main/java/alluxio/uri/MultiMasterAuthority.java

```diff
@@ -54,7 +58,11 @@ public boolean equals(Object o) {
       return false;
     }
     MultiMasterAuthority that = (MultiMasterAuthority) o;
-    return toString().equals(that.toString());
+
+    Set<String> firstAuthority = new HashSet<>(Arrays.asList(toString().split(",")));
+    Set<String> secondAuthority = new HashSet<>(Arrays.asList(that.toString().split(",")));
```

Comment:
How about cache these sets in a field for quicker comparison? Since authority objects are immutable, you can do this without worrying about data inconsistency between the set representation and the string representation.

Reply:
I'm not very comfortable with creating a temporary hash set, a list and an string array just for an equals method. Since `Authority` is part of `URI` and therefore part of `AlluxioURI`, and `AlluxioURI` is so ubiquitous across the code base, this can potentially bring a significant memory footprint at runtime. 

For example, the equals method is used by `AlluxioURI::isAncestorOf`:
https://github.com/Alluxio/alluxio/blob/8f74e6584b6d2d71be3a3737057808b6f77c44ae/core/common/src/main/java/alluxio/AlluxioURI.java#L455-L459

which is in turn used by `MountTable::reverseResolve`:
https://github.com/Alluxio/alluxio/blob/8f74e6584b6d2d71be3a3737057808b6f77c44ae/core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java#L316-L322

Can we do the equality check of authorities without allocating temporary objects?

Reply:
How about we first check whether the String is equal to the other String, and if not equal, we create sets to compare? So in most cases, this will not create objects.

Reply:
Made a commit here that compares authorities without allocating new temporary objects: https://github.com/dbw9580/alluxio/commit/3c2944d23e8734be55df3c6a98304174b91d147e
It seems I cannot just push changes into your repo, so that has to stay in my branch for now.